cmd/infra/powervs/destroy.go

```diff
@@ -206,6 +222,50 @@ func (options *DestroyInfraOptions) DestroyInfra(infra *Infra) (err error) {
 	return
 }
 
+// deleteDNSRecords deletes DNS records from CIS domain
+func deleteDNSRecords(options *DestroyInfraOptions) error {
+
+	if options.CisCRN == "" || options.CisDomainID == "" {
+		if options.BaseDomain == "" || options.Name == "" {
+			log(options.InfraID).Info("Skipping DNS record deletion. DNS Record deletion requires --name & --base-domain")
+			return nil
+		} else {
+			var err error
+			options.CisCRN, options.CisDomainID, err = getCisDomainDetails(options.BaseDomain)
+			if err != nil {
+				return fmt.Errorf("error retrieving cis domain details %w", err)
+			}
+		}
+	}
+
+	dnsRecordsV1, err := dnsrecordsv1.NewDnsRecordsV1(&dnsrecordsv1.DnsRecordsV1Options{Crn: &options.CisCRN, ZoneIdentifier: &options.CisDomainID, Authenticator: getIAMAuth()})
+	if err != nil {
+		return fmt.Errorf("error creating dns record service %w", err)
+	}
+
+	recordName := fmt.Sprintf("*.apps.%s.%s", options.Name, options.BaseDomain)
+	listDnsRecordsOpt := &dnsrecordsv1.ListAllDnsRecordsOptions{Name: &recordName}
+
+	dnsRecordsL, _, err := dnsRecordsV1.ListAllDnsRecords(listDnsRecordsOpt)
+	if err != nil {
+		return err
+	}
+
+	if len(dnsRecordsL.Result) <= 0 {
```

Comment:
just wondering if length every be less than 0 🤔

Reply:
modified to `len(dnsRecordsL.Result) < 1`, does this looks good? @mkumatag

Reply:
how about just `len(dnsRecordsL.Result) == 0`?

Reply:
modified🙂 @mkumatag